### PR TITLE
WELD-2404 support interception for InjectionTarget created

### PR DESCRIPTION
**Pull request by tremes**
programmatically for a MDB.

### CODE REVIEW

**Comment by tremes** [comment]
retest this please

File: modules/ejb/src/main/java/org/jboss/weld/module/ejb/EjbSupportImpl.java [hunk 1]
```diff
@@ -70,13 +73,25 @@ public void cleanup() {
...
             instantiator = new DefaultInstantiator<T>(type, null, manager);
         } else {
             // Session bean subclassed by the EJB container
             instantiator = SubclassedComponentInstantiator.forSubclassedEjb(type, implementationClass, null, manager);
         }
+        InterceptionModel interceptionModel = manager.getInterceptorModelRegistry().get(type.slim());
+        if (interceptionModel != null) {
+            if (interceptionModel.hasExternalNonConstructorInterceptors()) {
+                instantiator = SubclassedComponentInstantiator
+                        .forInterceptedDecoratedBean(implementationClass, null, (AbstractInstantiator<T>) instantiator, manager);
```
Comment (manovotn) [inline modules/ejb/src/main/java/org/jboss/weld/module/ejb/EjbSupportImpl.java]: I think the `null` value here is OK as the MDB is not a bean but a mere injection target. Also a quick glance at the code shows it takes `null` into consideration and CI doesn't complain.

File: modules/ejb/src/main/java/org/jboss/weld/module/ejb/EjbSupportImpl.java [hunk 2]
```diff
@@ -70,13 +73,25 @@ public void cleanup() {
...
+                instantiator = new InterceptorApplyingInstantiator<>(instantiator, interceptionModel, type.slim());
+
+            }
+            if (interceptionModel.hasExternalConstructorInterceptors()) {
```
Comment (manovotn) [inline modules/ejb/src/main/java/org/jboss/weld/module/ejb/EjbSupportImpl.java]: I wonder if `@AroundConstruct` makes any sense for MDB? We don't control it's lifecycle and don't really know what EJB container chooses to do with it and when. But perhaps I am missing something?

Comment (manovotn) [inline modules/ejb/src/main/java/org/jboss/weld/module/ejb/EjbSupportImpl.java]: Ok, we cleared this out - it makes sense but it's up to integrators which should use Weld SPI to enable this.

**Pull request merged**